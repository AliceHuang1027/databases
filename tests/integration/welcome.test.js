const {startServer,stopServer}=require('../../src/server')
const fetch = require('node-fetch')

describe('test welcome page',()=>{
    beforeAll(async ()=>{
        await startServer(20200)
    })
    afterAll(async ()=>{
        await stopServer()
    })
    test('should render welcome page correctly', async()=>{
        const result = await fetch('http://localhost:20200/').then(r=> r.text())
        expect(result).toMatchSnapshot()
    })
})
