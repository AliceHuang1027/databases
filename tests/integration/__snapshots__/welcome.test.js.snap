// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test welcome page should render databases page correctly 1`] = `
"<!--views/partials/head.ejs-->
<meta charset='UTF-8'>
<!--CSS (load bootstrap from a CDN)-->
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css\\">
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css\\">
<style>
    * {
        text-align: center;
    }
    input {
        width: 400px;
        height: 40px;
        margin: 5px;
    }
    button {
        width: 400px;
        height: 40px;
        margin: 10px;
    }
    .card {
        width: 450px;
        margin: auto;
    }
    .subtitle {
        margin: 10px;
    }
    .body {
        margin: 10px;
    }
</style>
<style>
    .db-card {
      border: solid 1px rgb(0,0,0);
      background-color: rgb(15,15,15);
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 10px;
      padding-bottom: 20px;
      border-radius: 5pX;
      cursor: pointer;
    }
    .db-card:hover {
      border: solid 1px rgb(10, 70, 120);
    }
</style>

<main>
    <div class=\\"card\\">
        <div class=\\"head\\">
            <h1 class=\\"title\\">Learn Databases</h1>
        </div>
        <div class=\\"body\\">
            <div class=\\"db-card postgres\\">
              <h3 class=\\"db-title\\">PostgreSQL</h3>
              <div class=\\"db-discription\\">A relational database management system emphasizing extensibility and technical standards compliance.</div>
            </div>
            <div class=\\"db-card mongodb\\">
              <h3 class=\\"db-title\\">MongoDB</h3>
              <div class=\\"db-discription\\">An open source database management system that uses a document-oriented database model which supports various forms of data.</div>
            </div>
            <div class=\\"db-card neo4j\\">
              <h3 class=\\"db-title\\">Neo4j</h3>
              <div class=\\"db-discription\\">A graph database management system that is the most papular graph database and the 22nd most popular database overall.</div>
            </div>
        </div>
        <div class=\\"tale\\">
        </div>
    </div>
</main>

<script>
    const postgres = document.querySelector('.postgres')
    const mongodb = document.querySelector('.mongodb')
    const neo4j = document.querySelector('.neo4j')

    postgres.addEventListener('click', () => {
      location.href = '/databases/postgres'
    })
    mongodb.addEventListener('click', () => {
      location.href = '/databases/mongodb'
    })
    neo4j.addEventListener('click', () => {
      location.href = '/databases/neo4j'
    })
</script>

"
`;

exports[`test welcome page should render setDBpassword page correctly 1`] = `
"<!--views/partials/head.ejs-->
<meta charset='UTF-8'>
<!--CSS (load bootstrap from a CDN)-->
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css\\">
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css\\">
<style>
    * {
        text-align: center;
    }
    input {
        width: 400px;
        height: 40px;
        margin: 5px;
    }
    button {
        width: 400px;
        height: 40px;
        margin: 10px;
    }
    .card {
        width: 450px;
        margin: auto;
    }
    .subtitle {
        margin: 10px;
    }
    .body {
        margin: 10px;
    }
</style>
<main>
    <div class=\\"card\\">
        <div class=\\"header\\">
            <h1 class=\\"title\\">Learn Databases</h1>
        </div>
        <div class=\\"body\\">
            <input class=\\"dbPassword\\" placeholder=\\"password\\">
            <button class=\\"setButton\\">Set</button>
        </div>
    </div>
</main>
<script>
    const dbPassword = document.querySelector('.dbPassword')
    const setButton = document.querySelector('.setButton')

    setButton.addEventListener('click',()=>{
        fetch('/api/users/:id',{
            method:\\"PATCH\\",
            headers:{
                'Content-Type': 'application/json'
            },
            body:JSON.stringify({
                password:dbPassword.value
            })
        }).then(r=>r.json()).then(r=>{
            return location.href = '/databases'
        })
    })
</script>
"
`;

exports[`test welcome page should render setPassword page correctly 1`] = `
"<!--views/partials/head.ejs-->
<meta charset='UTF-8'>
<!--CSS (load bootstrap from a CDN)-->
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css\\">
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css\\">
<style>
    * {
        text-align: center;
    }
    input {
        width: 400px;
        height: 40px;
        margin: 5px;
    }
    button {
        width: 400px;
        height: 40px;
        margin: 10px;
    }
    .card {
        width: 450px;
        margin: auto;
    }
    .subtitle {
        margin: 10px;
    }
    .body {
        margin: 10px;
    }
</style>
<main>
    <div class=\\"card\\">
        <div class=\\"head\\">
            <h1 class=\\"title\\">Learn Databases</h1>
        </div>
        <div class=\\"body\\">
            <div class=\\"subtitle\\">Set your password</div>
            <input class=\\"password1\\" placeholder=\\"create new password\\" type=\\"password\\"/><br>
            <input class=\\"password2\\" placeholder=\\"confirm password\\" type=\\"password\\"/>
            <button class=\\"submit\\">Submit</button>
        </div>
    </div>
</main>

<script>
    const path = window.location.pathname.split('/')
    const token = path.pop()
    const password1 = document.querySelector('.password1')
    const password2 = document.querySelector('.password2')
    const submit = document.querySelector('.submit')

    submit.addEventListener('click', () => {
        if (password1.value !== password2.value) {
            return alert(\\"Passwords do not match. Please check again.\\")
        }
        fetch('/api/passwordReset', {
            method: \\"POST\\",
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                token: token,
                password: password1.value
            })
        }).then(r => r.json()).then((r) => {
            if (r.error) return alert(r.error.message)
            return location.href = '/setDBPassword'
        }).catch((err) => {
            alert(\\"Setting password has failed. Please try again.\\", err)
        })
    })
</script>

"
`;

exports[`test welcome page should render sign in page correctly 1`] = `
"<!--views/partials/head.ejs-->
<meta charset='UTF-8'>
<!--CSS (load bootstrap from a CDN)-->
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css\\">
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css\\">
<style>
    * {
        text-align: center;
    }
    input {
        width: 400px;
        height: 40px;
        margin: 5px;
    }
    button {
        width: 400px;
        height: 40px;
        margin: 10px;
    }
    .card {
        width: 450px;
        margin: auto;
    }
    .subtitle {
        margin: 10px;
    }
    .body {
        margin: 10px;
    }
</style>
<main>
    <div class=\\"card\\">
        <div class=\\"head\\">
            <h1 class=\\"title\\">Learn Databases</h1>
        </div>
        <div class=\\"body\\">
            <div class=\\"subtitle\\">Please log in</div>
            <input class=\\"username\\" placeholder=\\"username\\" /><br>
            <input class=\\"password\\" placeholder=\\"password\\" type=\\"password\\"/>
            <button class=\\"submit\\">Submit</button>
        </div>
        <div class=\\"tale\\">
            <div>Forgot Password? <a href=\\"#\\">Find Password</a></div>
            <div>Don't have account? <a href=\\"/signup\\">Sign Up</a></div>
        </div>
    </div>
</main>

<script>
    const username = document.querySelector('.username')
    const password = document.querySelector('.password')
    const submit = document.querySelector('.submit')

    submit.addEventListener('click', () => {
        fetch('/api/session', {
            method: \\"POST\\",
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                username: username.value.toLowerCase(),
                password: password.value
            })
        }).then(r => r.json()).then((r) => {
            if (r.dbPassword) return location.href = '/setDBPassword' //path name is not decided yet  
            return location.href = '/databases' //path name is not decided yet
        })
    })
</script>

"
`;

exports[`test welcome page should render sign up page correctly 1`] = `
"<!--views/partials/head.ejs-->
<meta charset='UTF-8'>
<!--CSS (load bootstrap from a CDN)-->
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css\\">
<link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css\\">
<style>
    * {
        text-align: center;
    }
    input {
        width: 400px;
        height: 40px;
        margin: 5px;
    }
    button {
        width: 400px;
        height: 40px;
        margin: 10px;
    }
    .card {
        width: 450px;
        margin: auto;
    }
    .subtitle {
        margin: 10px;
    }
    .body {
        margin: 10px;
    }
</style>
<h1>Learn Databases</h1>

<p>Create Account</p>

<div>
  <input
    class=\\"email\\"
    type=\\"email\\"
    placeholder=\\"Email Address\\"
    required
    />
</div>

<div>
  <input
    class=\\"username\\"
    type=\\"text\\"
    placeholder=\\"Username\\"
    required
    />
</div>

<div>
  <button class=\\"submit\\">Create Account</button>
</div>

<div>
  Already have an account? <a href=\\"/signin\\">Login</a>
</div>

<script>
  const submit = document.querySelector('.submit');
  const check = document.querySelector('.check');
  const username = document.querySelector('.username')
  const email = document.querySelector('.email')

  submit.addEventListener('click', () => {
    fetch('/api/users', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        username: username.value.toLowerCase(),
        email: email.value.toLowerCase()
      })
    }).then(r => r.json()).then((res) => {
      if (res.error) {
        const message = res.error.message.split(':')
        alert(message[1].trim())
      } else {
        alert(\\"Check your email and set your password\\")
        return location.href = '/signin'
      }
    })
  })
</script>
"
`;

exports[`test welcome page should render welcome page correctly 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css\\"
    />
    <link
      rel=\\"stylesheet\\"
      href=\\"https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css\\"
    />

    <title>learndatabases.dev</title>
  </head>
  <body>
    <h1>Learn Databases</h1>
    <p><a href=\\"/signin\\">Login</a> / <a href=\\"/signup\\">Signup</a></p>

    <h2>Learn and try out different databases!</h2>
    <ul>
      <li>MongoDB, a NoSQL Database. <a href=\\"#\\">MongoDB</a></li>
      <li>Postgres, an SQL Database. <a href=\\"#\\">PostGresSQL</a></li>
      <li>Neo4j, a Graph Database. <a href=\\"#\\">Neo4j</a></li>
    </ul>

    <h1>Your Databases:</h1>

    <h3>Sponsors</h3>
    <ul>
      <li>
        <p><a href=\\"https://c0d3.com\\">C0d3</a> (They taught us to code)</p>
      </li>
    </ul>

    <p>Want to add a new Database or Sponsor? <a href=\\"#\\">Contact Us</a></p>

    <p>We collect anonymous analytics using Simple Analytics. View live: <a href=\\"#\\">simpleanalytics.com/learndatabases.dev</a></p>
  </body>
</html>
"
`;
